src/services/coreBackService.ts

```diff
@@ -144,3 +147,31 @@ export const getProvenIdentityUserDetails = (
     ),
   );
 };
+
+export const appVcReceived = async (req: Request): Promise<boolean> => {
+  try {
+    // Added for the browser tests
+    if (config.ENABLE_PREVIEW) {
+      return true;
+    }
+
+    const response = await axiosInstance.get(
+      config.API_CHECK_MOBILE_APP_VC_RECEIPT,
+      generateAxiosConfig(
+        `${config.API_BASE_URL}${config.API_CHECK_MOBILE_APP_VC_RECEIPT}`,
+        req,
+      ),
+    );
+
+    if (isJourneyResponse(response.data)) {
+      req.session.journey = response.data.journey;
+    }
+
+    return response.status === 200;
+  } catch (error) {
+    if (isAxiosError(error) && error.response?.status === 404) {
+      return false;
```

Comment:
Hmm I think it will be neater if we handle this in the caller i.e. [vc-receipt-status/middleware.ts](https://github.com/govuk-one-login/ipv-core-front/pull/1847/files#diff-30e30dd051e06c268b4f244c482cf4091d2f289fbf413a8d7e51614237389b22).
coreBackService probably shouldn't be setting things in the session as a side effect (it's really just a thin wrapper around the API) so I'd probably have this just return the core back response and handle the session and 404 error in the middleware, if this makes sense?